x/emissions/keeper/inference_synthesis/synth_palette_weight.go

```diff
@@ -36,14 +51,16 @@ func (p *SynthPalette) CalcWeightsGivenWorkers() (RegretInformedWeights, error)
 	}
 
 	normalizedForecasterRegrets := make(map[Worker]Regret)
-	for _, worker := range p.Forecasters {
-		regretFrac, err := p.ForecasterRegrets[worker].regret.Quo(stdDevRegretsPlusEpsilon)
-		if err != nil {
-			return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating regret fraction")
-		}
-		normalizedForecasterRegrets[worker] = regretFrac
-		if regretFrac.Gt(maxRegret) {
-			maxRegret = regretFrac
+	if len(p.ForecasterRegrets) > 0 {
+		for i, worker := range p.Forecasters {
+			regretFrac, err := p.ForecasterRegrets[worker].regret.Quo(stdDevRegretsPlusFTolerance)
+			if err != nil {
+				return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating regret fraction")
+			}
+			normalizedForecasterRegrets[worker] = regretFrac
+			if i == 0 || regretFrac.Gt(maxRegret) {
```

Comment:
I don't think we want this `i == 0` else we are re-initializing the maxRegret

Reply:
yes you are correct, idk why this got added back in 

cc @guilherme-brandao